code/modules/dwarven/dwarven_structures.dm

```diff
@@ -0,0 +1,145 @@
+		if(amountrefined < 1)
+			qdel(I)
+		else
+			new refined(drop_location(),amountrefined)
+			qdel(I)
+	. = ..()
+
+/obj/structure/destructible/dwarven/blood_pool
+	name = "Blood infuser"
+	icon_state = "blood_fountain"
+	var/charge_amount = 30
+
+
+/obj/structure/destructible/dwarven/blood_pool/attacked_by(obj/item/I, mob/living/user)
+	if(!istype(user,/mob/living/carbon/human))
+		return
+	var/mob/living/carbon/human/H = user
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand the instructions written in that ancient tongue</span>")
+		return
+	if(isbodypart(I))
+		to_chat(H, "<span class='notice'>The [I.name] turns to ash as you impale it on the bone. Infuser brightly flashes and blood pool swells.</span>")
+		charge_amount++
+		qdel(I)
+		return
+	. = ..()
+
+/obj/structure/destructible/dwarven/blood_pool/attack_hand(mob/user)
+	if(!istype(user,/mob/living/carbon/human))
+		return
+	var/mob/living/carbon/human/H = user
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand the instructions written in that ancient tongue</span>")
+		return
+
+	if(charge_amount >= 30)
+		to_chat(H, "<span class='notice'>You hear a faint whisper telling you to choose wiselty </span>")
+		var/choice
+		choice = alert(user,"You study the schematics etched into the stone...",,"Blitz rune","Air rune","Earth rune")
+		switch(choice)
+			if("Blitz rune")
+				new /obj/item/dwarven/rune_stone/blitz(get_turf(src))
+			if("Air rune")
+				new /obj/item/dwarven/rune_stone/air(get_turf(src))
+			if("Earth rune")
+				new /obj/item/dwarven/rune_stone/earth(get_turf(src))
+		to_chat(H, "<span class='notice'>You hear a whisper telling you you have chosen wisely.</span>")
+		charge_amount -= 20
+		return
+
+	. = ..()
+
+obj/structure/destructible/dwarven/mythril_press
+	name = "Ancient Mythril Press"
+	icon_state = "mythril_press"
+	var/list/obj/item/stack/sheet/mineral/loaded_mats = list()
+	var/list/obj/item/stack/sheet/mineral/needed_mats = list("diamond","uranium")
+
+obj/structure/destructible/dwarven/mythril_press/attacked_by(obj/item/I, mob/living/user)
+	if(istype(I,/obj/item/stack/sheet/mineral))
+		var/obj/item/stack/sheet/mineral/M = I
+		var/list/obj/item/stack/sheet/mineral/mat_list = list()
+		mat_list = loaded_mats & needed_mats
+		if(!(M in mat_list))
+			loaded_mats += "[M.name]"
+			M.amount--
+			to_chat(user, "<span class='notice'>You load [M.name] deep into the machine. There is no getting that sheet back.</span>")
+			if(M.amount == 0)
+				qdel(M)
+			return
+		to_chat(user, "<span class='notice'>[M.name] is already loaded into the machine!</span>")
+		return
+	. = ..()
+
+obj/structure/destructible/dwarven/mythril_press/attack_hand(mob/user)
+	var/list/obj/item/stack/sheet/mineral/mat_list = list()
+	mat_list = needed_mats & needed_mats
+	if(mat_list ~= needed_mats)
+		new /obj/item/stack/sheet/mineral/mythril(get_turf(src), 1)
```

Comment:
might be a balance/immershun concern since mythril gives rpg loot to items

Reply:
thats exactly why i chose mythril

Reply:
i chose mythril because miners want it, but only dwarves can operate the machinery making it an important decision. Do we let dwarves expand but trade with them for mythril or do we kill them?

Reply:
Might want to add a new material for that instead, I purposefully made mythril OP because its unobtanium tier.

Reply:
> Might want to add a new material for that instead, I purposefully made mythril OP because its unobtanium tier.

but i made mythril